From 34162d01fd781e1d19620def3080c415bdb609e0 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Tue, 6 Jul 2021 13:29:02 -0600
Subject: [PATCH] define APK_DEFAULT_ARCH for loongarch targets

Macros used have been taken from Loongson's work-in-progress GCC 12
port.
---
 src/apk_defines.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/apk_defines.h b/src/apk_defines.h
index 8901919..bc2863b 100644
--- a/src/apk_defines.h
+++ b/src/apk_defines.h
@@ -125,6 +125,12 @@ static inline int IS_ERR_OR_NULL(const void *ptr) { return IS_ERR(ptr) || !ptr;
 #define APK_DEFAULT_ARCH	"riscv32"
 #elif defined(__riscv) && __riscv_xlen == 64
 #define APK_DEFAULT_ARCH	"riscv64"
+#elif defined(__loongarch__) && defined(__loongarch32) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define APK_DEFAULT_ARCH	"loongarch32"
+#elif defined(__loongarch__) && defined(__loongarchx32) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define APK_DEFAULT_ARCH	"loongarchx32"
+#elif defined(__loongarch__) && defined(__loongarch64) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define APK_DEFAULT_ARCH	"loongarch64"
 #else
 #error APK_DEFAULT_ARCH not detected for this architecture
 #endif
-- 
2.27.0

